packages/@aws-cdk-testing/framework-integ/test/aws-route53/test/integ.cross-account-zone-delegation.ts

```diff
@@ -47,6 +48,12 @@ class ParentStack extends cdk.Stack {
     const parentZone = new route53.PublicHostedZone(this, 'HostedZone', {
       zoneName: parentZoneName,
     });
+    new route53.PrivateHostedZone(this, 'PrivateHostedZone', {
+      zoneName: parentZoneName,
+      vpc: new ec2.Vpc(this, 'TheVPC', {
+        ipAddresses: ec2.IpAddresses.cidr('10.0.0.0/16'),
+      }),
+    });
```

Comment:
might be missing something but how does this test the change thats implemented?

Reply:
The issue this PR fixes is the following:

When the parent account has 2 Hosted Zones with the same name (e.g. `abc.com`), and then when `CrossAccountZoneDelegationRecord` is deployed to a child account in order to delegate a subdomain (e.g. `xyz.abc.com`) to this child account, the `CrossAccountZoneDelegationRecord` handler/lambda would throw an error.

Therefore, I updated this integ test such that the parent account has one public and one private hosted zones with the same name. Then when the child account deploys `CrossAccountZoneDelegationRecord`, there shouldn't be any error with this PR fix.

Reply:
ah I see, so before the change this integ test with the private zone should've failed. Makes sense thanks for the explanation